src/main.rs

```diff
@@ -760,9 +764,15 @@ fn run_cli(cli: Cli) -> Result<()> {
             RecordIdState::InvalidRemoteId(err) => bail!("{err}"),
         },
         Command::Util { util_command } => match util_command {
-            UtilCommand::Check => {
+            UtilCommand::Check { fix } => {
                 info!("Validating record binary data and consistency, and checking for dangling records.");
-                record_db.validate()?;
+                let faults = record_db.recover(fix)?;
+                if !faults.is_empty() {
+                    error!("Database contains critical errors!");
```

Comment:
I think the message can be toned down a bit. Something like "Erroneous data found in the database."?